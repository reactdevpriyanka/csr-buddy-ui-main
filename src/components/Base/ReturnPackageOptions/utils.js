export const LABEL_CHOICES = [
  {
    label: '1 Label',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Labels',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Labels',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Labels',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Labels',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Labels',
    enabled: true,
    visible: true,
    value: 6,
  },
];

export const BOX_CHOICES = [
  {
    label: '1 Box',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Boxes',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Boxes',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Boxes',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Boxes',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Boxes',
    enabled: true,
    visible: true,
    value: 6,
  },
];
