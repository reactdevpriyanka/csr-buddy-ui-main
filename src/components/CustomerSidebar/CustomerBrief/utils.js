export const isWeight = (name) => name === 'WEIGHT_LIMIT';

export const blacklistedTags = new Set(['DISCOUNT_ABUSER', 'RETURNS_ABUSER', 'SUSPECTED_FRAUD']);
