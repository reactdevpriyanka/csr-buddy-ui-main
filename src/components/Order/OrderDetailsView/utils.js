import { getDayDateYearTimeTimezone } from '@/utils';
import { multiply } from 'lodash';

export const AllowableActions = {
  RETURN_ITEMS: 'RETURN_ITEMS',
  SEND_ORDER_INVOICE: 'SEND_ORDER_INVOICE',
  SEND_ORDER_CONFIRMATION: 'SEND_ORDER_CONFIRMATION',
  SEND_ORDER_CANCELED_NOTIFICATION: 'SEND_ORDER_CANCELED_NOTIFICATION',
  ALLOCATE_INVENTORY: 'ALLOCATE_INVENTORY',
  ADD_FULFILLMENT_COMMENT: 'ADD_FULFILLMENT_COMMENT',
  BLOCK_ORDER: 'BLOCK_ORDER',
  FORCE_RELEASE_ORDER: 'FORCE_RELEASE_ORDER',
  PROCESS_ORDER: 'PROCESS_ORDER',
  CANCEL_ORDER: 'CANCEL_ORDER',
  EDIT_SHIPPING_ADDRESS: 'EDIT_SHIPPING_ADDRESS',
  ADD_PROMOTION_CODE: 'ADD_PROMOTION_CODE',
  ADD_PRICE_ADJUSTMENT: 'ADD_PRICE_ADJUSTMENT',
  REMOVE_ITEM: 'REMOVE_ITEM',
  REDUCE_QUANTITY: 'REDUCE_QUANTITY',
  CREATE_PACKAGE: 'CREATE_PACKAGE',
  CONFIRM_SHIPMENT: 'CONFIRM_SHIPMENT',
  RESEND_SHIPMENT_MESSAGE: 'RESEND_SHIPMENT_MESSAGE',
  FORCE_CANCEL_RELEASE: 'FORCE_CANCEL_RELEASE',
  CANCEL_RELEASE: 'CANCEL_RELEASE',
  CANCEL_RETURN: 'CANCEL_RETURN',
  MARK_AS_RECEIVED: 'MARK_AS_RECEIVED',
  CREATE_NEW_LABELS: 'CREATE_NEW_LABELS',
};

export const AllowableActionTypes = {
  ORDER_ACTIONS: 'orderActions',
  RELEASE_ACTIONS: 'releaseActions',
  RETURN_ACTIONS: 'returnActions',
};

export const PackageReleaseTypes = Object.freeze({
  PACKAGE: Symbol('package'),
  RELEASE: Symbol('release'),
});

// format supplemental data attributes to for tooltip text
export const getAttributes = (tag, attributesMap, vetContactInfo, product, rowData, testId) => {
  switch (tag) {
    case 'DISCONTINUED':
      return (
        <span>
          {product?.discontinueDate != null && (
            <span data-testid="tag:discontinueDate" key="discontinueDate">
              <b>Discontinued Date:</b> {getDayDateYearTimeTimezone(product.discontinueDate)}
            </span>
          )}
        </span>
      );
    // case 'OUT_OF_STOCK': Disabling this until Force BackOrder and OOS back in stock dates are available in the api
    //   return (
    //     <span>
    //       {product?.discontinueDate != null && (
    //         <span key="discontinueDate">
    //           <b>Back in Stock Date:</b>
    //         </span>
    //       )}
    //     </span>
    //   );
    // case 'FORCED_BACK_ORDER':
    //   return (
    //     <span>
    //       {product?.discontinueDate != null && (
    //         <span key="discontinueDate">
    //           <b>Available Date:</b>
    //         </span>
    //       )}
    //     </span>
    //   );
    case 'CONNECT_WITH_A_VET':
      return attributesMap ? (
        <span>
          {vetContactInfo?.petName != null && (
            <span data-testid="tag:petName" key="petName">
              Pet Name: {vetContactInfo.petName}
              <br />
            </span>
          )}
          {attributesMap?.reason != null && (
            <span data-testid="tag:appointmentReason" key="appointmentReason">
              Appointment Reason: {attributesMap.reason}
              <br />
            </span>
          )}
          {attributesMap?.id != null && (
            <span data-testid="tag:appointmentId" key="appointmentId">
              Appointment ID: {attributesMap.id}
              <br />
            </span>
          )}
          {attributesMap?.endTime != null && (
            <span data-testid="tag:appointmentEndTime" key="appointmentEndTime">
              Appointment End Time: {`${getDayDateYearTimeTimezone(attributesMap.endTime)}`}
              <br />
            </span>
          )}
        </span>
      ) : null;
    case 'PHARMACY':
      return vetContactInfo ? (
        <span>
          {vetContactInfo?.clinicName != null && (
            <span data-testid="tag:clinicName" key="clinicName">
              Clinic Name: {vetContactInfo.clinicName}
              <br />
            </span>
          )}
          {vetContactInfo?.contactVet != null && (
            <span data-testid="tag:vetContacted" key="vetContacted">
              Vet Contacted: {vetContactInfo.contactVet ? 'Yes' : 'No'}
              <br />
            </span>
          )}
          {vetContactInfo?.vetProfileId != null && (
            <span data-testid="tag:vetProfileId" key="vetProfileId">
              Vet Profile ID: {vetContactInfo.vetProfileId}
              <br />
            </span>
          )}
          {vetContactInfo?.petName != null && (
            <span data-testid="tag:petName" key="petName">
              Pet Name: {vetContactInfo.petName}
              <br />
            </span>
          )}
        </span>
      ) : null;
    case 'DROPSHIP':
      return (
        <span>
          {rowData?.minDeliveryDate != null && rowData?.maxDeliveryDate != null && (
            <span data-testid="tag:minDeliveryDate" key="minDeliveryDate">
              <b>Delivers between:</b> {getDayDateYearTimeTimezone(rowData.minDeliveryDate)} and{' '}
              {getDayDateYearTimeTimezone(rowData.maxDeliveryDate)}
            </span>
          )}
        </span>
      );
    case 'PERSONALIZED':
      const entries = Object.entries(product?.personalizationAttributeMap || {}).sort(
        (obj1, obj2) => {
          return obj1[0].localeCompare(obj2[0]);
        },
      );
      return (
        <span>
          {entries?.map((entry) => {
            return (
              <div key={entry[0]}>
                <span data-testid={`tag:personalized:${entry[0]}`}>
                  <b>{entry[0]}:</b> {entry[1]}
                </span>
              </div>
            );
          })}
        </span>
      );
    default:
      return null;
  }
};

export const generateReturnItemKey = (returnItem) => returnItem.type + '_' + returnItem.state;

// Sums up the total quantity returned for each type of unique return
export const createTotalReturnedQuantities = (returnItems) => {
  const totalReturnedQuantities = {};

  for (const returnItem of returnItems) {
    const key = returnItem.lineItemId;
    const innerKey = generateReturnItemKey(returnItem);

    if (!totalReturnedQuantities[key]) {
      totalReturnedQuantities[key] = {};
    }

    if (!totalReturnedQuantities[key][innerKey]) {
      totalReturnedQuantities[key][innerKey] = 0;
    }

    // When either type of concession fails, the return service gives us back a totalCredit of 0 and no quantity field.
    // Because of this, we treat each concession like a totalCredit of 1 so it can be distributed to shipments like other concessions
    let specialCaseIncrement = 0;
    if (innerKey === 'PRODUCT_CONCESSION_FAILED' || innerKey === 'SHIPPING_CONCESSION_FAILED') {
      specialCaseIncrement = 1;
    }

    const increment =
      returnItem.type === 'PRODUCT_CONCESSION' || returnItem.type === 'SHIPPING_CONCESSION'
        ? returnItem?.totalCredit + specialCaseIncrement
        : returnItem?.quantity + specialCaseIncrement;

    totalReturnedQuantities[key][innerKey] += increment;
  }

  return totalReturnedQuantities;
};

export const orderHasItemsSpreadAcrossShipments = (shipments, currShipment, itemMap = {}) => {
  if (shipments?.length === 1) return false;

  const lineItemIds = Object.keys(itemMap);
  for (const lineItemId of lineItemIds) {
    const otherShipmentLineItemIds = shipments
      ?.filter((shipment) => shipment.id !== currShipment.id)
      ?.map((shipment) => shipment?.shipmentItems)
      ?.flat()
      ?.map(({ lineItemId }) => lineItemId);
    if (otherShipmentLineItemIds?.includes(lineItemId)) return true;
  }
  return false;
};

// When the same lineItemId is found across multiple shipments, we have no way of knowing which returnItems (if any)
// are attributed to any specific item in any specific shipment.
//
// The number of returns processed for a lineItemId will always be less than the total quantity of that lineItemId across all shipments
// so we iterate through the shipments untill we run out of return tags for that lineItemId.
export const assignReturnsToShipments = (shipments, returnItems, itemMap = {}) => {
  const totalReturnedQuantities = createTotalReturnedQuantities(returnItems);

  for (const [index, shipment] of shipments.entries()) {
    for (const shipmentItem of shipment.shipmentItems) {
      const lineItemId = shipmentItem.lineItemId;

      if (totalReturnedQuantities[lineItemId]) {
        const returnTagsToRender = Object.entries(totalReturnedQuantities[lineItemId])
          ?.filter(([returnItemKey, quantityReturned]) => quantityReturned >= 0.001)
          ?.map(([returnItemKey, quantityReturned]) => returnItemKey);

        shipments[index] = {
          ...shipment,
          returnTagsToRender: returnTagsToRender,
        };

        for (const returnItemKey in totalReturnedQuantities[lineItemId]) {
          const returnedQuantity = totalReturnedQuantities[lineItemId][returnItemKey];
          if (returnedQuantity < 0) continue; // No need to continue to compute if there is no returnable quantity left for a lineItemId

          const returnItem = returnItems.find(
            (item) => generateReturnItemKey(item) === returnItemKey,
          );

          if (!returnItem) continue;

          let quantityToDecrement;

          switch (returnItem?.type) {
            case 'PRODUCT_CONCESSION':
              quantityToDecrement = multiply(
                shipmentItem?.quantity * Number.parseFloat(itemMap[lineItemId]?.unitPrice?.value),
              );
              break;
            case 'SHIPPING_CONCESSION':
              quantityToDecrement = returnItem?.totalCredit;
              break;
            case 'REPLACEMENT':
            case 'REFUND':
              quantityToDecrement = shipmentItem?.quantity;
              break;
            default:
              quantityToDecrement = 0;
              break;
          }

          // Subtract the decrement quantity from the quantity of items (or total value of products in concession scenarios)
          // returned to only add returns to be rendered to a shipment if there were enough returns to justify it
          // when the same lineItemId is spread across multiple packages
          totalReturnedQuantities[lineItemId][returnItemKey] =
            returnedQuantity - quantityToDecrement;
        }
      }
    }
  }
};
