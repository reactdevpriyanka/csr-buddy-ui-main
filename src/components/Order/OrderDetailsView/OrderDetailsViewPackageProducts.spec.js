import { renderWrap } from '@/utils';
import useCustomer from '@/hooks/useCustomer';
import useOrder from '@/hooks/useOrder';
import customer from '__mock__/customers/nessa';
import lineItems from '__mock__/orderdetails/orderdetails-line-items';
import currPackage from '__mock__/orderdetails/orderdetails-package-data';
import * as features from '@/features';
import OrderDetailsViewPackageProducts from './OrderDetailsViewPackageProducts';

jest.mock('@/hooks/useCustomer');
jest.mock('@/hooks/useOrder');
jest.mock('@/hooks/useAgentInteractions');

describe('<OrderDetailsViewPackageProducts />', () => {
  const defaultProps = {
    orderNumber: '1504724130',
    packageNum: 1,
    lineItems: lineItems,
    currPackage: currPackage,
    status: 'SHIPPED',
    isActionAllowed: () => true,
  };

  useCustomer.mockReturnValue({
    data: customer,
    error: null,
  });
  useOrder.mockReturnValue({
    quantity: '2',
    error: null,
  });

  const render = renderWrap(OrderDetailsViewPackageProducts, { defaultProps });

  test('it should render children', () => {
    expect(render()).toMatchSnapshot();
  });

  test('it should display the OrderDetailsViewPackageProductsContainer', () => {
    const { getByTestId } = render();
    expect(getByTestId(`orderDetailsViewPackageProductsContainer`)).toBeTruthy();
    expect(getByTestId(`orderDetailsViewPackageProductsStatus`)).toHaveTextContent(
      defaultProps.status,
    );
  });

  test('it should display the items quantity minus sign', () => {
    const { getByTestId } = render();
    expect(getByTestId(`gwf:multi-item-quantity-minus`)).toBeTruthy();
  });

  test('it should display the remove item trash icon', () => {
    const { getByTestId } = render();
    expect(getByTestId(`remove-item-trash-icon`)).toBeTruthy();
  });

  test('should display item status', () => {
    jest.spyOn(features, 'useFeature').mockImplementation(() => true);
    const { queryByText } = render();
    expect(queryByText(defaultProps.status)).toBeInTheDocument();
  });

  test('should not display item status', () => {
    jest.spyOn(features, 'useFeature').mockImplementation(() => false);
    const { queryByText } = render();
    expect(queryByText(defaultProps.status)).not.toBeInTheDocument();
  });

  describe('when isActionAllowed equals false', () => {
    const defaultProps = {
      orderNumber: '1504724130',
      packageNum: 1,
      lineItems: lineItems,
      currPackage: currPackage,
      isActionAllowed: () => false,
    };

    useCustomer.mockReturnValue({
      data: customer,
      error: null,
    });

    const render = renderWrap(OrderDetailsViewPackageProducts, { defaultProps });

    test('it should not display the items quantity minus sign', () => {
      const { queryByTestId } = render();
      expect(queryByTestId(`gwf:multi-item-quantity-minus`)).not.toBeInTheDocument();
    });

    test('it should not display the remove item trash icon', () => {
      const { queryByTestId } = render();
      expect(queryByTestId(`remove-item-trash-icon`)).not.toBeInTheDocument();
    });
  });
});
