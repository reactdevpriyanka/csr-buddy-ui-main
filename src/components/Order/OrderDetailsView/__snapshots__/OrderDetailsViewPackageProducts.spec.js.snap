// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetailsViewPackageProducts /> it should render children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1o4yg1q-MuiPaper-root"
        data-testid="orderDetailsViewPackageProductsContainer"
      >
        <div
          class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        >
          <table
            aria-label="sticky table"
            class="MuiTable-root MuiTable-stickyHeader css-12vjeh8-MuiTable-root"
          >
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  style="min-width: 75px; max-width: 75px;"
                >
                  <a
                    aria-label="ITEM #114674"
                    class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                    data-testid="product:catalogEntryId:link:114674"
                    href="http://www-tst.chewy.com/app/dp/114674"
                    rel="noopener"
                    target="_blank"
                  >
                    <figure
                      class="makeStyles-figure-6"
                      data-testid="product:thumbnail"
                    >
                      <div
                        class="makeStyles-container-7"
                      >
                        <img
                          alt=""
                          class="makeStyles-thumbnail-22"
                          src="https://img.chewy.com/is/image/catalog/86899_MAIN._SS250_.jpg"
                        />
                      </div>
                    </figure>
                  </a>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  style="min-width: 500px; max-width: 500px; padding: 0px 16px 0px 16px;"
                >
                  <div
                    class="makeStyles-description-8"
                  >
                    <a
                      aria-label="ITEM #114674"
                      class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                      data-testid="product:catalogEntryId:link:114674"
                      href="http://www-tst.chewy.com/app/dp/114674"
                      rel="noopener"
                      target="_blank"
                    >
                      <span
                        class="makeStyles-partNumber-9"
                        data-testid="product:catalogEntryId:114674"
                      >
                        ITEM #86899
                      </span>
                    </a>
                    <a
                      aria-label="ITEM #114674"
                      class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                      data-testid="product:catalogEntryId:link:114674"
                      href="http://www-tst.chewy.com/app/dp/114674"
                      rel="noopener"
                      target="_blank"
                    >
                      <span>
                        Bones & Chews Made in USA Roasted Marrow Bone 6" Dog Treat
                      </span>
                    </a>
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-0.5 css-1neqpi-MuiGrid-root"
                    >
                      <span>
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.1 makeStyles-quantityPanel-19 css-1cu0x8w-MuiGrid-root"
                        >
                          <span
                            class="makeStyles-quantityOrdered-26"
                          >
                            Qty Ordered
                            : 
                            5
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                  style="min-width: 175px; max-width: 175px; padding: 0px 16px 0px 16px;"
                >
                  <div
                    class="makeStyles-root-32"
                  >
                    <div
                      class="makeStyles-selector-33"
                    >
                      <button
                        class=""
                        data-testid="gwf:multi-item-quantity-minus"
                      />
                      <span
                        class="makeStyles-quantity-36"
                      >
                        5
                      </span>
                      <button
                        class="disabled"
                        data-testid="gwf:multi-item-quantity-plus"
                      />
                    </div>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  style="min-width: 100px; padding: 50px 16px 16px 16px;"
                >
                  <div
                    class="makeStyles-tableComponent-41"
                  >
                    <span
                      class="makeStyles-label-38"
                    >
                      Status
                    </span>
                    <span
                      class="makeStyles-text-39"
                      data-testid="orderDetailsViewPackageProductsStatus"
                      style="color: rgb(0, 107, 43);"
                    >
                      <div
                        class="makeStyles-itemStatus-43"
                      >
                         
                        <span>
                          SHIPPED
                        </span>
                         
                        
                      </div>
                      <br />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
                >
                  <div
                    class="makeStyles-tableComponent-41"
                  >
                    <span
                      class="makeStyles-label-38"
                    >
                      Unit $
                    </span>
                    <span
                      class="makeStyles-text-39"
                    >
                      $5.47
                       
                      <br />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                  style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
                >
                  <div
                    class="makeStyles-tableComponent-41"
                  >
                    <span
                      class="makeStyles-label-38"
                    >
                      Discount
                    </span>
                    <span
                      class="makeStyles-text-39"
                    >
                      $0.00
                       
                      <br />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                  style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
                >
                  <div
                    class="makeStyles-tableComponent-41"
                  >
                    <span
                      class="makeStyles-label-38"
                    >
                      Total
                    </span>
                    <span
                      class="makeStyles-text-39"
                    >
                      $27.35
                       
                      <br />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                  style="min-width: 60px; max-width: 60px; padding: 0px 16px 0px 16px;"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    data-testid="remove-item-trash-icon"
                    tabindex="0"
                    type="button"
                    variant="outlined"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-trashIcon-12 css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteOutlineSharpIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 21h12V7H6v14zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="makeStyles-packagesFooter-17"
          >
            <span
              class="makeStyles-label-14"
            >
              Package 1 total   
            </span>
            <span
              class="makeStyles-text-15"
            >
              $NaN
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1o4yg1q-MuiPaper-root"
      data-testid="orderDetailsViewPackageProductsContainer"
    >
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root MuiTable-stickyHeader css-12vjeh8-MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                style="min-width: 75px; max-width: 75px;"
              >
                <a
                  aria-label="ITEM #114674"
                  class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                  data-testid="product:catalogEntryId:link:114674"
                  href="http://www-tst.chewy.com/app/dp/114674"
                  rel="noopener"
                  target="_blank"
                >
                  <figure
                    class="makeStyles-figure-6"
                    data-testid="product:thumbnail"
                  >
                    <div
                      class="makeStyles-container-7"
                    >
                      <img
                        alt=""
                        class="makeStyles-thumbnail-22"
                        src="https://img.chewy.com/is/image/catalog/86899_MAIN._SS250_.jpg"
                      />
                    </div>
                  </figure>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                style="min-width: 500px; max-width: 500px; padding: 0px 16px 0px 16px;"
              >
                <div
                  class="makeStyles-description-8"
                >
                  <a
                    aria-label="ITEM #114674"
                    class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                    data-testid="product:catalogEntryId:link:114674"
                    href="http://www-tst.chewy.com/app/dp/114674"
                    rel="noopener"
                    target="_blank"
                  >
                    <span
                      class="makeStyles-partNumber-9"
                      data-testid="product:catalogEntryId:114674"
                    >
                      ITEM #86899
                    </span>
                  </a>
                  <a
                    aria-label="ITEM #114674"
                    class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
                    data-testid="product:catalogEntryId:link:114674"
                    href="http://www-tst.chewy.com/app/dp/114674"
                    rel="noopener"
                    target="_blank"
                  >
                    <span>
                      Bones & Chews Made in USA Roasted Marrow Bone 6" Dog Treat
                    </span>
                  </a>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-0.5 css-1neqpi-MuiGrid-root"
                  >
                    <span>
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.1 makeStyles-quantityPanel-19 css-1cu0x8w-MuiGrid-root"
                      >
                        <span
                          class="makeStyles-quantityOrdered-26"
                        >
                          Qty Ordered
                          : 
                          5
                        </span>
                      </div>
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                style="min-width: 175px; max-width: 175px; padding: 0px 16px 0px 16px;"
              >
                <div
                  class="makeStyles-root-32"
                >
                  <div
                    class="makeStyles-selector-33"
                  >
                    <button
                      class=""
                      data-testid="gwf:multi-item-quantity-minus"
                    />
                    <span
                      class="makeStyles-quantity-36"
                    >
                      5
                    </span>
                    <button
                      class="disabled"
                      data-testid="gwf:multi-item-quantity-plus"
                    />
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                style="min-width: 100px; padding: 50px 16px 16px 16px;"
              >
                <div
                  class="makeStyles-tableComponent-41"
                >
                  <span
                    class="makeStyles-label-38"
                  >
                    Status
                  </span>
                  <span
                    class="makeStyles-text-39"
                    data-testid="orderDetailsViewPackageProductsStatus"
                    style="color: rgb(0, 107, 43);"
                  >
                    <div
                      class="makeStyles-itemStatus-43"
                    >
                       
                      <span>
                        SHIPPED
                      </span>
                       
                      
                    </div>
                    <br />
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
              >
                <div
                  class="makeStyles-tableComponent-41"
                >
                  <span
                    class="makeStyles-label-38"
                  >
                    Unit $
                  </span>
                  <span
                    class="makeStyles-text-39"
                  >
                    $5.47
                     
                    <br />
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
              >
                <div
                  class="makeStyles-tableComponent-41"
                >
                  <span
                    class="makeStyles-label-38"
                  >
                    Discount
                  </span>
                  <span
                    class="makeStyles-text-39"
                  >
                    $0.00
                     
                    <br />
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                style="min-width: 100px; max-width: 100px; padding: 0px 16px 0px 16px;"
              >
                <div
                  class="makeStyles-tableComponent-41"
                >
                  <span
                    class="makeStyles-label-38"
                  >
                    Total
                  </span>
                  <span
                    class="makeStyles-text-39"
                  >
                    $27.35
                     
                    <br />
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                style="min-width: 60px; max-width: 60px; padding: 0px 16px 0px 16px;"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  data-testid="remove-item-trash-icon"
                  tabindex="0"
                  type="button"
                  variant="outlined"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-trashIcon-12 css-i4bv87-MuiSvgIcon-root"
                      data-testid="DeleteOutlineSharpIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 21h12V7H6v14zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="makeStyles-packagesFooter-17"
        >
          <span
            class="makeStyles-label-14"
          >
            Package 1 total   
          </span>
          <span
            class="makeStyles-text-15"
          >
            $NaN
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
